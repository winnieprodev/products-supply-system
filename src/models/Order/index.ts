export * from "./enums";
export * from "./Order.d";
