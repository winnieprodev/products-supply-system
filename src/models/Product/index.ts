export * from "./Product.d";
export * from "./enums";
