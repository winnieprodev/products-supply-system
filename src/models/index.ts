export * from "./Cart";
export * from "./Order";
export * from "./Outlet";
export * from "./Product";
export * from "./Search";
export * from "./Staff";
export * from "./Supplier";
export * from "./User";
export * from "./utils";
