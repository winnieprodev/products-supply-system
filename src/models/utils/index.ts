export * from "./async.d";
export * from "./enums";
