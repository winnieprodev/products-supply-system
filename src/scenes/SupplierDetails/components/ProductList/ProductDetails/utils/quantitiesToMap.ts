import { Map } from "immutable";
import { CartQuantity, QuantityType } from "models";

export const quantitiesToMap = (quantities: CartQuantity[]) => {
  return Map<QuantityType, number>(
    quantities.map(({ type, amount }) => [type, amount] as const) as Iterable<
      [QuantityType, number]
    >,
  );
};
