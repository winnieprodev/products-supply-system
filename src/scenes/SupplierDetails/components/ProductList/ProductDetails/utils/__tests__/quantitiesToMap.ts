import { QuantityType } from "models";
import { quantitiesToMap } from "../quantitiesToMap";

describe("ProductDetails: utils", () => {
  describe("quantitiesToMap", () => {
    test("return proper value", () => {
      expect(quantitiesToMap([]).toJS()).toMatchSnapshot();
      expect(
        quantitiesToMap([
          { type: QuantityType.WHOLE, amount: 3 },
          { type: QuantityType.LOOSE, amount: 1 },
        ]).toJS(),
      ).toMatchSnapshot();
    });
  });
});
