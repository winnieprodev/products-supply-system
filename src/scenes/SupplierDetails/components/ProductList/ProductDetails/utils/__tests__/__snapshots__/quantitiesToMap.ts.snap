// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductDetails: utils quantitiesToMap return proper value 1`] = `Object {}`;

exports[`ProductDetails: utils quantitiesToMap return proper value 2`] = `
Object {
  "loose": 1,
  "whole": 3,
}
`;
